### DISCUSSION TITLE
Multiple sources in one xml file

### DISCUSSION THREAD

**Discussion by keystrom-laplace** [iptv-org/community#1125] [Q&A]
I wonder if you can use multiple EPG sources for an EPG xml file. If it's possible, then can anyone show how to do it? Thank you.

**Comment by ghost** [comment]
Look at my fork. I just added two python scripts, the first one collects several xml files for each source, in my case there are 3 of them `'tvpassport.com', 'tv.blue.ch', 'tvtv.us'`. After that, from those 3 files with channels I create 3 files with epg respectively using github workflow. after that i use the second script to merge those 3 files with epg into one and archive it. Everything looks great and works but there is a problem. if you add some other sources like `'flixed.io'` then during the assembly of the EPG some symbols inside the links are not escaped, such as `&` and maybe something else. because of this the players think that this xml file is invalid and the second python script cannot combine these files. I don't know how to fix this. I think this is the problem of the grabber itself


![image](https://github.com/user-attachments/assets/ab327943-9934-4f86-8c61-8deece2694a6)

**Comment by freearhey** [comment]
https://github.com/iptv-org/epg?tab=readme-ov-file#use-custom-channel-list
